azure-iot-device/samples/async-pnp-samples/sample.py

```diff
@@ -26,158 +28,154 @@
+        self.telemetry.humid = Telemetry()
+        self.properties.state = Property()
+        self.commands.blink = Command()
+        self.commands.turnOff = Command()
+        self.commands.turnOn = Command()
+        self.commands.runDiagnostics = Command()
+        self.properties.name = WriteableProperty()
+        self.properties.brightness = WriteableProperty()
+
+
+class DeviceInformation(BaseInterface):
     def __init__(self, interface_instance_name):
         super(DeviceInformation, self).__init__(
             interface_instance_name, "urn:azureiot:DeviceInformation:1"
         )
-        self.manufacturer = Property()
-        self.model = Property()
-        self.sw_version = Property()
-        self.os_name = Property()
-        self.processor_architecture = Property()
-        self.processor_manufacturer = Property()
-        self.total_storage = Property()
-        self.total_memory = Property()
+        self.properties.manufacturer = Property()
+        self.properties.model = Property()
+        self.properties.swVersion = Property()
+        self.properties.osName = Property()
+        self.properties.processorArchitecture = Property()
+        self.properties.processorManufacturer = Property()
+        self.properties.totalStorage = Property()
+        self.properties.totalMemory = Property()
 
 
-class SampleExit(Interface):
+class SampleExit(BaseInterface):
     def __init__(self, interface_instance_name):
         super(SampleExit, self).__init__(
-            interface_instance_name, "urn:azureiotsdknode:SampleInterface:SampleExit:1"
+            interface_instance_name, "urn:azureiotsdknode:SampleBaseInterface:SampleExit:1"
         )
-        self.exit = Command()
+        self.commands.exit = Command()
 
 
-class ModelDefinition(Interface):
+class ModelDefinition(BaseInterface):
     def __init__(self, interface_instance_name):
         super(ModelDefinition, self).__init__(
             interface_instance_name, "urn:azureiot:ModelDiscovery:ModelDefinition:1"
         )
-        self.get_model_definition = Command()
+        self.commands.getModelDefinition = Command()
 
 
 environmental_sensor = EnvironmentalSensor("environmentalSensor")
 device_information = DeviceInformation("deviceInformation")
 model_definition = ModelDefinition("urn_azureiot_ModelDiscovery_ModelDefinition")
-exit_interface = SampleExit("urn_azureiotsdknode_SampleInterface_SampleExit")
+exit_interface = SampleExit("urn_azureiotsdknode_SampleBaseInterface_SampleExit")
 
 
 # Note: this follows the method pattern from the pythonIoTHubDeviceClient object.
-async def environmental_command_listener(interface):
+async def environmental_model_listener(interface):
     while True:
-        command_request = await interface.receive_pnp_command()
+        incoming_message = await interface.receive_pnp_message()
 
-        if command_request.command_name == "blink":
+        if incoming_message.target == interface.commands.blink:
```

Comment:
>if incoming_message.target == interface.commands.blink: [](start = 8, length = 55)

Note: the incoming_message object could be a CommandRequest or it could be a PropertyUpdateRequest.  They both have a target attribute (and probably come from the same base class?)